src/gpio/gpio.c

```diff
@@ -73,6 +75,14 @@ mraa_gpio_init_internal(mraa_adv_func_t* func_table, int pin)
     dev->advance_func = func_table;
     dev->pin = pin;
 
+    // GPIO sysfs path may store the default path names schema or any other specific to vendor
+    dev->gpio_path = (char*) calloc(MAX_SIZE, sizeof(char));
+    if (dev->gpio_path == NULL) {
+        syslog(LOG_CRIT, "gpio%i: Failed to allocate memory for context's attribute", pin);
```

Comment:
I'd still recommend to elaborate on "context's attribute" and just state directly it's the gpio_path we are talking about.